oap-server/analyzer/agent-analyzer/src/main/java/org/apache/skywalking/oap/server/analyzer/provider/trace/parser/TraceAnalyzer.java

```diff
@@ -49,30 +49,26 @@ public void doAnalysis(SegmentObject segmentObject) {
 
         createSpanListeners();
 
-        try {
-            notifySegmentListener(segmentObject);
-
-            segmentObject.getSpansList().forEach(spanObject -> {
-                if (spanObject.getSpanId() == 0) {
-                    notifyFirstListener(spanObject, segmentObject);
-                }
-
-                if (SpanType.Exit.equals(spanObject.getSpanType())) {
-                    notifyExitListener(spanObject, segmentObject);
-                } else if (SpanType.Entry.equals(spanObject.getSpanType())) {
-                    notifyEntryListener(spanObject, segmentObject);
-                } else if (SpanType.Local.equals(spanObject.getSpanType())) {
-                    notifyLocalListener(spanObject, segmentObject);
-                } else {
-                    log.error("span type value was unexpected, span type name: {}", spanObject.getSpanType()
-                                                                                              .name());
-                }
-            });
-
-            notifyListenerToBuild();
-        } catch (Throwable e) {
-            log.error(e.getMessage(), e);
-        }
```

Comment:
Why do you remove try/catch? Just expose the error to handler?

Reply:
https://github.com/apache/skywalking/blob/f54f639c4d9207c20490519121eeabe6acf37374/oap-server/server-receiver-plugin/skywalking-trace-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/trace/provider/handler/v8/grpc/TraceSegmentReportServiceHandler.java#L72-L78

> Just expose the error to handler?

Yes, otherwise `errorCounter` is incorrect.

Reply:
Remember to add this kind of fix(bug) in the changlog, which you have submitted yet.